src/compressed_tensors/quantization/lifecycle/forward.py

```diff
@@ -304,6 +308,42 @@ def wrapped_forward(self, *args, **kwargs):
     setattr(module, "forward", bound_wrapped_forward)
 
 
+def wrap_module_forward_quantized_attn(module: Module, scheme: QuantizationScheme):
+    # expects a module already initialized and injected with the parameters in
+    # initialize_module_for_quantization
+    if hasattr(module.forward, "__func__"):
+        forward_func_orig = module.forward.__func__
+    else:
+        forward_func_orig = module.forward.func
+
+    @wraps(forward_func_orig)  # ensures docstring, names, etc are propagated
+    def wrapped_forward(self, *args, **kwargs):
+
+        past_key_value = scheme.output_activations.get_kv_cache()
```

Comment:
Why is get_kv_cache on the scheme's output_activations

Reply:
This is a good comment. To change, we'll need to introduce kv_cache to the quantization scheme, which is more appropriate. Then we will need to add logic to extract and save it to the appropriate part of the config. 

Depending on how strongly you feel, this is a cosmetic change, so I wouldnt put too much effort